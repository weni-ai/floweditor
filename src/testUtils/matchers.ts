import { commaListsOr } from 'common-tags';
import { Exit } from 'flowTypes';
import { RenderNode } from 'store/flowContext';

const snapshot = require('jest-snapshot');

const matchers: jest.ExpectExtendMap = {};

interface MatchResult {
  message: () => string;
  pass: boolean;
}

function toBeUnique<T>(this: jest.MatcherUtils, received: any[]): MatchResult {
  const seen: { [value: string]: boolean } = {};

  for (const item of received) {
    const stringified = JSON.stringify(item, null, 1);
    if (seen[stringified]) {
      return {
        message: () => `Duplicate item in array:\n${stringified}`,
        pass: false,
      };
    } else {
      seen[stringified] = true;
    }
  }

  return {
    message: () => 'Array is unique',
    pass: true,
  };
}

function toMatchCallSnapshot<T>(
  this: jest.MatcherUtils,
  received: any,
  snapshotName?: string,
): MatchResult {
  return snapshot.toMatchSnapshot.call(
    this,
    received.mock.calls[0],
    snapshotName,
  );
}

function toHaveInboundConnections<T>(
  this: jest.MatcherUtils,
  received: RenderNode,
): MatchResult {
  if (Object.keys(received.inboundConnections).length > 1) {
    return {
      message: () => `Node ${received.node.uuid} has an inbound connection`,
      pass: true,
    };
  }

  return {
    message: () => `Node ${received.node.uuid} has no inbound connections`,
    pass: false,
  };
}

function toHaveInboundFrom<T>(
  this: jest.MatcherUtils,
  received: RenderNode,
  expected: Exit,
): MatchResult {
  for (const exitUUID of Object.keys(received.inboundConnections)) {
    if (exitUUID === expected.uuid) {
      return {
        message: () =>
          `Node ${received.node.uuid} incorrectly has inbound from ${expected.uuid}`,
        pass: true,
      };
    }
  }

  return {
    message: () =>
      `Node ${received.node.uuid} doesn't have inbound connection from ${expected.uuid}`,
    pass: false,
  };
}

function toPointTo<T>(
  this: jest.MatcherUtils,
  received: Exit,
  expected: RenderNode,
): MatchResult {
  if (received.destination_uuid === expected.node.uuid) {
    return {
      message: () =>
        `${received.uuid} incorrectly points to ${expected.node.uuid}`,
      pass: true,
    };
  }

  return {
    message: () =>
      `Expected ${received.uuid} to point to ${expected.node.uuid}`,
    pass: false,
  };
}

function toHaveExitWithDestination<T>(
  this: jest.MatcherUtils,
  received: RenderNode,
): MatchResult {
  for (const exit of received.node.exits) {
    if (exit.destination_uuid !== null) {
      return {
        message: () =>
          `Exit ${exit.uuid} in node ${received.node.uuid} has a destination`,
        pass: true,
      };
    }
  }

  return {
    message: () => `All exits for ${received.node.uuid} are disconnected`,
    pass: false,
  };
}

function toHaveExitThatPointsTo<T>(
  this: jest.MatcherUtils,
  received: RenderNode,
  expected: RenderNode,
): MatchResult {
  for (const exit of received.node.exits) {
    if (exit.destination_uuid === expected.node.uuid) {
      return {
        message: () =>
          `${exit.uuid} incorrectly has exit that points to ${expected.node.uuid}`,
        pass: true,
      };
    }
  }

  return {
    message: () =>
      `Expected ${received.node.uuid} to have an exit that points to ${expected.node.uuid}`,
    pass: false,
  };
}

function toHavePayload<T>(
  this: jest.MatcherUtils,
  store: any,
  actionType: string,
  expectedPayload: any,
): MatchResult {
  const payload = JSON.stringify(expectedPayload);
  for (const actionTaken of store.getActions()) {
    if (actionTaken.type === actionType) {
      if (JSON.stringify(actionTaken.payload) === payload) {
        return {
          message: () =>
            `Result contained action type ${actionType} with payload ${payload}`,
          pass: true,
        };
      }
    }
  }

  return {
    message: () =>
      `Could not find action type ${actionType} with payload ${payload}`,
    pass: false,
  };
}

function toHaveReduxActions<T>(
  this: jest.MatcherUtils,
  store: any,
  actionTypes: string[],
): MatchResult {
  const missedTypes: any[] = [];
  for (const actionTaken of store.getActions()) {
    for (const actionType of actionTypes) {
      if (actionTaken.type === actionType) {
        return {
          message: () => `Result contained action type ${actionType}`,
          pass: true,
        };
      } else {
        missedTypes.push(actionType);
      }
    }
  }

  return {
    message: () =>
      commaListsOr`Could not find action type(s): ${[...missedTypes]}`,
    pass: false,
  };
}

expect.extend({
  toPointTo,
  toHaveExitThatPointsTo,
  toHaveInboundFrom,
  toHaveExitWithDestination,
  toHaveInboundConnections,
  toHavePayload,
  toHaveReduxActions,
  toMatchCallSnapshot,
  toBeUnique,
});
