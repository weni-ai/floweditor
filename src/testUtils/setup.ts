/* eslint-disable no-redeclare */
import { Console } from 'console';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Exit } from 'flowTypes';
import { RenderNode } from 'store/flowContext';

// Declare custom matcher types
declare global {
  namespace jest {
    interface Matchers<R> {
      toPointTo(renderNode: RenderNode): R;
      toHaveExitThatPointsTo(renderNode: RenderNode): R;
      toHaveInboundFrom(exit: Exit): R;
      toHaveExitWithDestination(): R;
      toHaveInboundConnections(): R;
      toHavePayload(action: string, payload: any): R;
      toHaveReduxActions(actions: string[]): R;
      toMatchCallSnapshot(snapshotName?: string): R;
    }
  }
}

// Ensure console logs are visible while running tests https://github.com/facebook/jest/issues/3853
global.console = new Console(process.stderr, process.stderr);

// Configure Enzyme adapter
configure({ adapter: new Adapter() });

// RAF shim
// tslint:disable-next-line:ban-types
(global as any).requestAnimationFrame = (callback: Function) => {
  setTimeout(callback, 0);
};
