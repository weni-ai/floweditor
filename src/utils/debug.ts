import { FlowStoreProps } from 'components/flow/Flow';
import { DebugState } from 'store/editor';

import mutate from 'immutability-helper';

/* istanbul ignore next -- @preserve */
export default class Debug {
  private props: FlowStoreProps;
  private state: DebugState;

  constructor(props: FlowStoreProps, initial: DebugState) {
    this.props = props;
    this.state = initial || { showUUIDs: false };
  }

  public showUUIDs(): DebugState {
    const updated = mutate(this.state, { $merge: { showUUIDs: true } });
    this.props.mergeEditorState({ debug: updated });
    return updated;
  }
}
