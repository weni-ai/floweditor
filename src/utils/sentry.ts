import * as Sentry from '@sentry/react';

const initSentry = (dsn: string) => {
  Sentry.init({
    dsn,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.httpClientIntegration({
        failedRequestStatusCodes: [[400, 499], [500, 599]],
      }),
      Sentry.replayIntegration(),
      Sentry.replayCanvasIntegration(),
    ],
    tracesSampleRate: 1.0,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  });
};

export default initSentry;
