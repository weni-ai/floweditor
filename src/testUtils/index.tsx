// TODO: Remove use of Function
// tslint:disable:ban-types
import { ConfigProviderContext, fakePropType } from 'config/ConfigProvider';
import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import { FlowDefinition, FlowEditorConfig } from 'flowTypes';
import mutate, { Query } from 'immutability-helper';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import createStore from 'store/createStore';
import { RenderNodeMap } from 'store/flowContext';
import { getFlowComponents } from 'store/helpers';
import AppState, { initialState } from 'store/state';
import config from 'test/config';
import * as matchers from 'testUtils/matchers';
import { merge, set } from 'utils';

// we need to use require syntax to bust implicit any
const boring: FlowDefinition = require('test/flows/boring.json');

// force our matchers to be read in
const match = matchers;

export interface Resp {
  assets: { [key: string]: any }[];
}

export interface QueryString {
  [key: string]: string;
}

export const contextTypes: { [key: string]: Function } = {
  config: fakePropType,
  store: fakePropType,
  assetService: fakePropType,
};

export const baseState: AppState = mutate(initialState, {
  flowContext: merge({
    definition: require('test/flows/colors.json') as FlowDefinition,
  }),
  editorState: merge({
    language: { id: 'eng', name: 'English' },
  }),
});

const flowEditorConfig: FlowEditorConfig = config;

export const configProviderContext: ConfigProviderContext = {
  config: flowEditorConfig,
};

export const setMock = (
  implementation?: (...args: any[]) => any,
): Query<jest.Mock> => set(jest.fn(implementation));

/**
 * Compose setup method for component tests
 */
export const composeSetup = <P extends {}>(
  Component: React.ComponentClass | React.SFC,
  baseProps: P = {} as any,
  baseDuxState: AppState | Partial<AppState> = baseState,
  baseContext: ConfigProviderContext = configProviderContext,
) => (
  shallowRender: boolean = true,
  propOverrides: Query<P | Partial<P>> = {},
  duxStateOverrides: Query<AppState | Partial<AppState>> = {},
  contextOverrides: Query<
    ConfigProviderContext | Partial<ConfigProviderContext>
  > = {},
  childContextTypeOverrides: { [key: string]: Function } = {},
  children: (JSX.Element | React.ComponentClass | React.SFC)[] = [],
) => {
  const props = mutate(baseProps, propOverrides);
  const store = createStore(mutate(
    baseDuxState,
    duxStateOverrides,
  ) as AppState);
  let context = mutate(baseContext, merge({ store }));

  if (Object.keys(duxStateOverrides).length > 0) {
    context = mutate(context, duxStateOverrides);
  }

  if (Object.keys(contextOverrides).length > 0) {
    context = mutate(context, contextOverrides as ConfigProviderContext);
  }

  const childContextTypes: { [contextProp: string]: Function } = mutate(
    contextTypes,
    childContextTypeOverrides,
  );
  // tslint:disable-next-line:ban-types
  const wrapper = (shallowRender ? (shallow as Function) : (mount as Function))(
    <Component {...props}>{children}</Component>,
    {
      context,
      childContextTypes,
    },
  );

  return {
    wrapper,
    props,
    context,
    instance: wrapper.instance(),
  };
};

export const composeSpy = (obj: Record<string, any> | React.ComponentClass) => (
  instanceMethod: string,
) =>
  jest.spyOn(
    (obj as React.ComponentClass).prototype || obj,
    instanceMethod as any,
  );

/**
 * Wait for promises in queue to resolve
 */
export const flushPromises = () =>
  new Promise((resolve: any) => setImmediate(resolve));

/**
 * Restore spy mocks (distinct from mocks created w/ jest.fn()).
 * Use to declaratively restore multiple spy mocks,
 * otherwise just call .mockRestore() on lone spy.
 */
export const restoreSpies = (...spies: jest.SpyInstance<any>[]) =>
  spies.forEach((spy: any) => spy.mockRestore());

// To-do: type this method's output, can pass it prop generic ingested by getComponentTestUtils
/**
 * NOTE: borrowed from EventBrite: https://github.com/eventbrite/javascript/blob/master/react/testing.md#finding-nodes
 * Finds all instances of components in the rendered `componentWrapper` that are DOM components
 * with the `data-spec` attribute matching `name`.
 * @param {ReactWrapper|ShallowWrapper} componentWrapper - Rendered componentWrapper (result of mount, shallow, or render)
 * @param  {string} snacName - Name of `data-spec` attribute value to find
 * @param {string|Function} (Optional) typeFilter - (Optional) Expected type of the wrappers (defaults to all HTML tags)
 * @returns {ReactWrapper|ReactWrapper[]|ShallowWrapper|ShallowWrapper[]} Matching DOM components
 */
export const getSpecWrapper = (
  componentWrapper: ReactWrapper<{}, {}> | ShallowWrapper<{}, {}>,
  specName: string,
  attributeName: string = 'data-spec',
): any => componentWrapper.find(`[${attributeName}="${specName}"]`);

export const composeDuxState = (
  query: Query<AppState | Partial<AppState>> = {},
  duxState = baseState,
) => mutate(duxState, query);

export const composeComponentTestUtils = <P extends {}>(
  Component: React.ComponentClass | React.SFC | any,
  baseProps: P = {} as any,
  baseDuxState: AppState | Partial<AppState> = baseState,
  baseContext: ConfigProviderContext = configProviderContext,
) => ({
  setup: composeSetup<P>(Component, baseProps, baseDuxState, baseContext),
  spyOn: composeSpy(Component),
});

export const prepMockDuxState = (): {
  testNodes: RenderNodeMap;
  mockDuxState: AppState;
} => {
  const testNodes = getFlowComponents(boring).renderNodeMap;

  return {
    testNodes,
    mockDuxState: {
      ...initialState,
      flowContext: {
        ...initialState.flowContext,
        definition: boring,
        nodes: testNodes,
      },
    },
  };
};

export const createMockStore: Function = configureStore([thunk]);

export const mock = <T extends {}, K extends keyof T>(
  object: T,
  property: K,
  value: T[K],
) => {
  Object.defineProperty(object, property, { get: () => value });
};
