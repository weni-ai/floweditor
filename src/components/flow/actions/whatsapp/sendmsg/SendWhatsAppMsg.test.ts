import { SendWhatsAppMsg } from 'flowTypes';
import { composeComponentTestUtils } from 'testUtils';
import { createSendWhatsAppMsgAction } from 'testUtils/assetCreators';
import { set } from 'utils';

import SendWhatsAppMsgComp from 'components/flow/actions/whatsapp/sendmsg/SendWhatsAppMsg';

const sendWhatsAppMsgAction = createSendWhatsAppMsgAction();

const { setup } = composeComponentTestUtils<SendWhatsAppMsg>(
  SendWhatsAppMsgComp,
  sendWhatsAppMsgAction,
);

describe(SendWhatsAppMsgComp.name, () => {
  describe('render', () => {
    it('should render text prop when passed', () => {
      const { wrapper, props } = setup(false, { text: set('Text') });

      expect(wrapper.text()).toBe(props.text);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render quick replies list when passed', () => {
      const { wrapper } = setup(false, {
        quick_replies: set(['Quick Reply 1']),
      });

      expect(wrapper.text()).toContain('Quick Reply 1');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render list items when passed', () => {
      const { wrapper } = setup(false, {
        list_items: set([{ title: 'Title 1', description: 'Description 1' }]),
      });

      expect(wrapper.text()).toContain('Title 1');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render attachment items when passed', () => {
      const { wrapper } = setup(false, {
        attachment: set(true),
      });

      expect(wrapper).toMatchSnapshot();
    });
  });
});
