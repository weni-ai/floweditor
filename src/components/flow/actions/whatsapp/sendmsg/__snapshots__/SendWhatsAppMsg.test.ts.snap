// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendWhatsAppMsgComp render should render list items when passed 1`] = `
<SendWhatsAppMsgComp
  attachment={null}
  button_text=""
  footer=""
  header_text=""
  header_type="media"
  interaction_type="list"
  list_items={
    Array [
      Object {
        "description": "Description 1",
        "title": "Title 1",
      },
    ]
  }
  quick_replies={Array []}
  text=""
  type="send_msg"
  uuid="b1f332f3-bdd3-4891-aec5-1843a712dbf1"
>
  <div />
  <div
    className=""
  />
  <div
    className="summary"
  >
    <div
      className="list_items"
    >
      <Pill
        advanced={true}
        disabled={true}
        key="b1f332f3-bdd3-4891-aec5-1843a712dbf1Title 1"
        maxLength={20}
        text="Title 1"
      >
        <ForwardRef
          clickable={false}
          data-advanced={true}
          disabled={true}
          text="Title 1"
        >
          <ForwardRef
            clickable={false}
            component={
              Object {
                "_Ctor": Object {
                  "0": [Function],
                },
                "_compiled": true,
                "computed": Object {
                  "currentComponent": [Function],
                },
                "name": "unnnic-tag",
                "props": Object {
                  "clickable": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "count": Object {
                    "default": 0,
                    "type": [Function],
                  },
                  "disabled": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "enableTooltip": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "hasBackButton": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "hasCloseIcon": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "icon": Object {
                    "default": null,
                    "type": [Function],
                  },
                  "scheme": Object {
                    "default": "aux-purple",
                    "type": [Function],
                  },
                  "text": Object {
                    "default": null,
                    "type": [Function],
                  },
                  "tooltipText": Object {
                    "default": null,
                    "type": [Function],
                  },
                  "type": Object {
                    "default": "default",
                    "type": [Function],
                    "validator": [Function],
                  },
                },
                "render": [Function],
                "staticRenderFns": Array [],
              }
            }
            data-advanced={true}
            disabled={true}
            text="Title 1"
            vuereact-combined-options={
              Object {
                "react": Object {},
                "vue": Object {},
              }
            }
          >
            <r
              clickable={false}
              component={
                Object {
                  "_Ctor": Object {
                    "0": [Function],
                  },
                  "_compiled": true,
                  "computed": Object {
                    "currentComponent": [Function],
                  },
                  "name": "unnnic-tag",
                  "props": Object {
                    "clickable": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "count": Object {
                      "default": 0,
                      "type": [Function],
                    },
                    "disabled": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "enableTooltip": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "hasBackButton": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "hasCloseIcon": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "icon": Object {
                      "default": null,
                      "type": [Function],
                    },
                    "scheme": Object {
                      "default": "aux-purple",
                      "type": [Function],
                    },
                    "text": Object {
                      "default": null,
                      "type": [Function],
                    },
                    "tooltipText": Object {
                      "default": null,
                      "type": [Function],
                    },
                    "type": Object {
                      "default": "default",
                      "type": [Function],
                      "validator": [Function],
                    },
                  },
                  "render": [Function],
                  "staticRenderFns": Array [],
                }
              }
              data-advanced={true}
              disabled={true}
              text="Title 1"
              vuereact-combined-options={
                Object {
                  "react": Object {
                    "componentWrap": "div",
                    "componentWrapAttrs": Object {
                      "__use_react_component_wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                    "slotWrap": "div",
                    "slotWrapAttrs": Object {
                      "__use_react_slot_wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                  },
                  "vue": Object {
                    "componentWrapAttrs": Object {
                      "data-use-vue-component-wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                    "componentWrapHOC": [Function],
                    "slotWrapAttrs": Object {
                      "data-use-vue-slot-wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                  },
                }
              }
            >
              <Component
                portals={Array []}
              >
                <div
                  data-use-vue-component-wrap=""
                  style={
                    Object {
                      "all": "unset",
                    }
                  }
                >
                  <div
                    key="null"
                  />
                </div>
              </Component>
            </r>
          </ForwardRef>
        </ForwardRef>
      </Pill>
    </div>
  </div>
</SendWhatsAppMsgComp>
`;

exports[`SendWhatsAppMsgComp render should render quick replies list when passed 1`] = `
<SendWhatsAppMsgComp
  attachment={null}
  button_text=""
  footer=""
  header_text=""
  header_type="media"
  interaction_type="list"
  list_items={Array []}
  quick_replies={
    Array [
      "Quick Reply 1",
    ]
  }
  text=""
  type="send_msg"
  uuid="b1f332f3-bdd3-4891-aec5-1843a712dbf1"
>
  <div />
  <div
    className="summary"
  >
    <div
      className="quick_replies"
    >
      <Pill
        advanced={true}
        disabled={true}
        key="b1f332f3-bdd3-4891-aec5-1843a712dbf1Quick Reply 1"
        maxLength={20}
        text="Quick Reply 1"
      >
        <ForwardRef
          clickable={false}
          data-advanced={true}
          disabled={true}
          text="Quick Reply 1"
        >
          <ForwardRef
            clickable={false}
            component={
              Object {
                "_Ctor": Object {
                  "0": [Function],
                },
                "_compiled": true,
                "computed": Object {
                  "currentComponent": [Function],
                },
                "name": "unnnic-tag",
                "props": Object {
                  "clickable": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "count": Object {
                    "default": 0,
                    "type": [Function],
                  },
                  "disabled": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "enableTooltip": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "hasBackButton": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "hasCloseIcon": Object {
                    "default": false,
                    "type": [Function],
                  },
                  "icon": Object {
                    "default": null,
                    "type": [Function],
                  },
                  "scheme": Object {
                    "default": "aux-purple",
                    "type": [Function],
                  },
                  "text": Object {
                    "default": null,
                    "type": [Function],
                  },
                  "tooltipText": Object {
                    "default": null,
                    "type": [Function],
                  },
                  "type": Object {
                    "default": "default",
                    "type": [Function],
                    "validator": [Function],
                  },
                },
                "render": [Function],
                "staticRenderFns": Array [],
              }
            }
            data-advanced={true}
            disabled={true}
            text="Quick Reply 1"
            vuereact-combined-options={
              Object {
                "react": Object {},
                "vue": Object {},
              }
            }
          >
            <r
              clickable={false}
              component={
                Object {
                  "_Ctor": Object {
                    "0": [Function],
                  },
                  "_compiled": true,
                  "computed": Object {
                    "currentComponent": [Function],
                  },
                  "name": "unnnic-tag",
                  "props": Object {
                    "clickable": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "count": Object {
                      "default": 0,
                      "type": [Function],
                    },
                    "disabled": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "enableTooltip": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "hasBackButton": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "hasCloseIcon": Object {
                      "default": false,
                      "type": [Function],
                    },
                    "icon": Object {
                      "default": null,
                      "type": [Function],
                    },
                    "scheme": Object {
                      "default": "aux-purple",
                      "type": [Function],
                    },
                    "text": Object {
                      "default": null,
                      "type": [Function],
                    },
                    "tooltipText": Object {
                      "default": null,
                      "type": [Function],
                    },
                    "type": Object {
                      "default": "default",
                      "type": [Function],
                      "validator": [Function],
                    },
                  },
                  "render": [Function],
                  "staticRenderFns": Array [],
                }
              }
              data-advanced={true}
              disabled={true}
              text="Quick Reply 1"
              vuereact-combined-options={
                Object {
                  "react": Object {
                    "componentWrap": "div",
                    "componentWrapAttrs": Object {
                      "__use_react_component_wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                    "slotWrap": "div",
                    "slotWrapAttrs": Object {
                      "__use_react_slot_wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                  },
                  "vue": Object {
                    "componentWrapAttrs": Object {
                      "data-use-vue-component-wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                    "componentWrapHOC": [Function],
                    "slotWrapAttrs": Object {
                      "data-use-vue-slot-wrap": "",
                      "style": Object {
                        "all": "unset",
                      },
                    },
                  },
                }
              }
            >
              <Component
                portals={Array []}
              >
                <div
                  data-use-vue-component-wrap=""
                  style={
                    Object {
                      "all": "unset",
                    }
                  }
                >
                  <div
                    key="null"
                  />
                </div>
              </Component>
            </r>
          </ForwardRef>
        </ForwardRef>
      </Pill>
    </div>
  </div>
  <div
    className=""
  />
</SendWhatsAppMsgComp>
`;

exports[`SendWhatsAppMsgComp render should render text prop when passed 1`] = `
<SendWhatsAppMsgComp
  attachment={null}
  button_text=""
  footer=""
  header_text=""
  header_type="media"
  interaction_type="list"
  list_items={Array []}
  quick_replies={Array []}
  text="Text"
  type="send_msg"
  uuid="b1f332f3-bdd3-4891-aec5-1843a712dbf1"
>
  <div>
    <div
      className="line"
      key="b1f332f3-bdd3-4891-aec5-1843a712dbf10"
    >
      Text
    </div>
  </div>
  <div
    className=""
  />
  <div
    className=""
  />
</SendWhatsAppMsgComp>
`;
